### ISSUE TITLE
ENSO coefficient maps large non-zero means

### ISSUE THREAD

**Issue by brianhenn** [ai2cm/ace#434]
TLDR: the `EnsoCoefficient` aggregator needs to ensure that the Nino3.4 time series interpolated to the data actually has zero mean, which I don't think is currently happening.

For some variables and evaluation time periods, the `EnsoCoefficient` aggregator produces coefficient maps that have large non-zero mean values. When this happens, the spatial pattern of the maps still seems physically correct, and it affects the prediction and target equally such that the difference metric is not affected, but it is distracting. In particular, temperature variables seem prone to this, I think because they have the largest mean values. e.g., [this map](https://wandb.ai/ai2cm/ace/reports/inference-enso_coefficient-coefficient_maps-TMP2m-24-08-23-14-42-07---Vmlldzo5MTMxOTgy) of the TMP2m ENSO coefficient from an ~80-year AMIP run has a global mean value of about -5K per K of Nino3.4, which doesn't make physical sense, even though the correct patterns are evident. 

I don't want to just enforce that the coefficient maps have zero mean by removing it, because there are physical reasons that they could have a mean: 1) ENSO phase does change the global mean eg for temperature, and 2) the Nino3.4 index also has longer-term trends that may correlate with global trends.

But I suspect a large amount of the mean values in the maps are coming from the fact that we use a running sum approach to compute the coefficients that requires that the predictor have a mean of zero. This is [imperfectly enforced by the current code](https://github.com/ai2cm/full-model/blob/7109285ba1f2d58fa632e39dcadc50a8c19e2852/fme/fme/core/aggregator/inference/enso/enso.py#L338-L392), which does take the mean out of the 3-monthly Nino3.4 index, but does it before that index is interpolated to the high-temporal frequency prediction/target data. My hypothesis is that this leaves a non-zero mean in the interpolated index and that this produces a lot of the non-zero mean in the maps. 

We could either: 1) compute initially what the mean of the Nino3.4 index interpolated to the full run data time coordinate will be and remove it, or 2) track running sums of the index and the variable over the run, and remove their product.
